src/Microsoft.Health.Client/Authentication/AuthenticationOptions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Client.Authentication;
+
+public class AuthenticationOptions
+{
+    public const string SectionName = "Authentication";
+
+    public bool Enabled { get; set; }
+
+    public AuthenticationType? AuthenticationType { get; set; }
```

Comment:
Do we need `AuthenticationType`  if we have options for each type below?
Eg. `if (OAuth2ClientCertificateCredential != null) { ... }`

Reply:
I think I prefer being explicit here. My reasoning is that without being explicit we're implying some sort of order of precedence.